Added on to the docs for splits using encoders (#414)
KMKfw/kmk_firmware: docs/scanners.md

```

> 

Reviewer: ```suggestion
```

        # etc...
    ]
```
#### Multiple Scanners coord_mapping and keymap changes

> To add more scanners you need to add onto your "coord_mapping" Example below.

Reviewer: ```suggestion
To add more scanners you need to add onto your `coord_mapping`.

Example:
```

> Coord_mapping before with just "MatrixScanner" on a 58 key split keyboard:

Reviewer: ```suggestion
`coord_mapping` with just one `MatrixScanner` on a 58 key split keyboard:
```

```python
coord_mapping = [
     0,  1,  2,  3,  4,  5,         35, 34, 33, 32, 31, 30,
     6,  7,  8,  9, 10, 11,         41, 40, 39, 38, 37, 36,
    12, 13, 14, 15, 16, 17,         47, 46, 45, 44, 43, 42,
    18, 19, 20, 21, 22, 23, 29, 59, 53, 52, 51, 50, 49, 48,
            25, 26, 27, 28,         58, 57, 56, 55, 
    ]
```

> Coord_mapping after using "RotaryioEncoder" and "MatrixScanner" on the same 58 key split keyboard with an encoder on each half:

Reviewer: ```suggestion
`coord_mapping` using `MatrixScanner` and `RotaryioEncoder` on the same 58 key split keyboard with an encoder on each half:
```

I took the liberty to remove the "before/after" adjectives, because it's not necessarily a temporal relationship, rather a with-or-without thing.
Also, just to be on the safe side, I'd put the scanners in the order they are in the matrix list (the order is important).

```python
coord_mapping = [
     0,  1,  2,  3,  4,  5,         37, 36, 35, 34, 33, 32,
     6,  7,  8,  9, 10, 11,         43, 42, 41, 40, 39, 38,
    12, 13, 14, 15, 16, 17,         49, 48, 47, 46, 45, 44,
    18, 19, 20, 21, 22, 23, 29, 61, 55, 54, 53, 52, 51, 50,
            25, 26, 27, 28,         60, 59, 58, 57,
            30, 31,                         62, 63 
    ]
```

> On the top left side of a standard split keyboard "coord_mapping", right below that you see a split keyboard where "RotaryioEncoder" and "MatrixScanner"(the default scanner) are used. In the before example, we used to count from 0 to 29 while the top right side starts at 30. With the addition of the encoder scanner, the left side has 2 additional keys making it count up to 31 and the right side would then start at 32 and count to 63. This means that keys 30, 31, 62, and 63 are for encoders. Notice that all of the encoders are at the end of the array. Therefore, we need to add 4 more key codes to the end of our `keyboard.keymap`, They will be used for the encoders.

Reviewer: ```suggestion
On the top left side of a standard split keyboard `coord_mapping`, right below that you see a split keyboard where `RotaryioEncoder` and `MatrixScanner` (the default scanner) are used.
In the single scanner example, we used to count from 0 to 29 while the top right side starts at 30.
With the addition of the encoder scanner, the left side has 2 additional keys making it count up to 31 and the right side would then start at 32 and count to 63.
This means that keys 30, 31, 62, and 63 are for encoders.
Notice that all of the encoders are at the end of the array, because we put the encoder scanner after the matrix scanner in `keyboard.matrix`.
Therefore, we need to add 4 more key codes in the corresponding places of our `keyboard.keymap`, they will be used for the encoders.
```

Newline after every full-stop. Makes it a bit easier to edit/see were the sentences end, and doesn't show up in the rendered text.
Plus some pedantic detailing.

        # etc...
    ]
```
#### Multiple Scanners coord_mapping and keymap changes

> To add more scanners you need to add onto your your " coord_mapping " Example below.

Reviewer: use proper inline code quoting

> Before:

Reviewer: instead of "before/after" use something more descriptive, like "here's a `coord_mapping` for a split with single 30-key scanner", "and here's the same split with two scanners, the 30-key matrix and an encoder".

```python
coord_mapping = [
     0,  1,  2,  3,  4,  5,         35, 34, 33, 32, 31, 30,
     6,  7,  8,  9, 10, 11,         41, 40, 39, 38, 37, 36,
    12, 13, 14, 15, 16, 17,         47, 46, 45, 44, 43, 42,
    18, 19, 20, 21, 22, 23, 29, 59, 53, 52, 51, 50, 49, 48,
            25, 26, 27, 28,         58, 57, 56, 55, 
    ]
```

After:
```python
coord_mapping = [
     0,  1,  2,  3,  4,  5,         37, 36, 35, 34, 33, 32,
     6,  7,  8,  9, 10, 11,         43, 42, 41, 40, 39, 38,
    12, 13, 14, 15, 16, 17,         49, 48, 47, 46, 45, 44,
    18, 19, 20, 21, 22, 23, 29, 61, 55, 54, 53, 52, 51, 50,
            25, 26, 27, 28,         60, 59, 58, 57,
            30, 31,                         62, 63 
    ]
```

> This is as complicated as it can be. Split with "RotaryioEncoder" and "MatrixScanner". You will see that the top left side used to count up to 29 and the right side started at 30. With the encoder module added on the left side has 2 more keys so it now counts up to 31 and the right side starts its count at 32 and continues to 63. 30,31,62 and 63 are all for encoders. You will see that all of the encoders are at the end of the array, this means that we need to add 4 more key codes to the end of our `keymap` in `main.py`  

Reviewer: * again, use proper inline code.
* one sentence per line
* -> "end of our `keyboard.keymap`." , pointing to `main.py` doesn't make sense.

> 
> 

Reviewer: ```suggestion
```
